Src/FluentAssertions/Events/EventMonitor.cs

```diff
@@ -117,20 +121,48 @@ public void Dispose()
     {
         foreach (EventRecorder recorder in recorderMap.Values)
         {
-            recorder.Dispose();
+            DisposeSafeIfRequested(recorder);
         }
 
         recorderMap.Clear();
     }
 
+    private void DisposeSafeIfRequested(EventRecorder recorder)
+    {
+        try
+        {
+            recorder.Dispose();
+        }
+        catch when (options.ShouldIgnoreEventAccessorExceptions)
+        {
+            // ignore
+        }
+    }
+
     private void AttachEventHandler(EventInfo eventInfo, Func<DateTime> utcNow)
     {
         if (!recorderMap.TryGetValue(eventInfo.Name, out _))
         {
             var recorder = new EventRecorder(subject.Target, eventInfo.Name, utcNow, threadSafeSequenceGenerator);
             if (recorderMap.TryAdd(eventInfo.Name, recorder))
             {
-                recorder.Attach(subject, eventInfo);
+                AttachEventHandler(eventInfo, recorder);
+            }
+        }
+    }
+
+    private void AttachEventHandler(EventInfo eventInfo, EventRecorder recorder)
+    {
+        try
+        {
+            recorder.Attach(subject, eventInfo);
+        }
+        catch when (options.ShouldIgnoreEventAccessorExceptions)
+        {
+            if (!recorderMap.TryRemove(eventInfo.Name, out _))
+            {
+                throw new InvalidOperationException(
+                    $"Could not remove event {eventInfo.Name} with broken event accessor from event recording.");
             }
```

Comment:
For my understanding, why do we need this?
I tried to remove it and no tests failed.

I get that the `Try` part is hard to exercise because of concurrency, but I guess(?) that not removing from `recorderMap` at should be testable?

Reply:
@apazureck Have you had a chance to look into this?

Reply:
Hi, sorry for the long delay, I somehow overlooked this message before I was on holiday the last weeks.

You are right, I did not test this explicitly, as it is internal and I cannot provoke the error without manipulating the internal state of the `EventMonitor` via reflection.

I thought it would be good to have an exception, if some code will change and this will be removed in the meantime. I cannot think of any valid reason to do this, but the person, who did the change, should be alerted, that this is not an expected behavior. I thought this is better than silently swallow the inconsistent behavior.

But I can write another test this week manipulating the eventmonitor, or I can remove this exception. Tell me what you think is the better approach.

Reply:
Great to have you back 😄

Not sure if we're in sync here.

1)
Why are we removing the event from `recorderMap` in the first place?

I removed these lines and no tests failed which indicates that there's a scenario we're not covering.
```cs
if (!recorderMap.TryRemove(eventInfo.Name, out _))
{
    throw new InvalidOperationException(
        $"Could not remove event {eventInfo.Name} with broken event accessor from event recording.");
}
```

This can be tested with something like this.
```cs
var subject = new TestEventBrokenEventHandlerRaising();
using var monitor = subject.Monitor<IAddFailingEvent>(opt => opt.IgnoreEventAccessorExceptions());
monitor.GetRecordingFor(nameof(IAddFailingEvent.AddFailingEvent))
```

If we remove the event from `recorderMap`, then `GetRecordingFor` throws an `InvalidOperationException` that we're not monitoring `IAddFailingEvent`, otherwise we get a empty recording.
What do we want here?

I haven't used these monitoring APIs in my day job, so I'm not _that_ familiar with all the ins and outs and what is expected here.

2)
This question is only relevant if the outcome of 1) is to keep `TryRemove`.

As I understand `TryRemove` (just had to look it up) a false means that:
* The value was already removed when calling `TryRemove`
* Another concurrent call removed the value, 
  * I.e. for N concurrent calls to `TryRemove` on an existing key, 1 returns true and N-1 return false.

In `AttachEventHandler` we don't throw an exception if `TryAdd` returns false, so I don't think we should throw here.

Reply:
@dennisdoomen can you help here, since you know more about Events than me